### ISSUE TITLE
[a11y]: Form control with "slider" role has no associated label

### ISSUE THREAD

**Issue by youngjunekoh** [carbon-design-system/carbon#12046]
### Package

carbon-components-react

### Browser

Chrome

### Operating System

MacOS

### Package version

v7.56.0

### React version

v17.0.2

### Automated testing tool and ruleset

IBM equal access accessibility checker - latest

### Description

IBM equal access accessibility checker is flagging a slider component as needing an associated label (slider thumb specifically). 

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-slider--default

### Steps to reproduce

Run a scan (IBM equal access accessibility checker plugin) on the page listed above to see the violation.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
